### DISCUSSION TITLE
Disable InterfaceStubGenerator

### DISCUSSION THREAD

**Discussion by MeikelLP** [reactiveui/refit#1764] [Q&A] [answered]
Any way to disable InterfaceStubGenerator ?

It generates broken code (missing method implementations) beacuse my interfaces come from another source generator but I also have partial code which extends these generated code.

Sadly it's currently impossible to exclude this analyzer.... https://github.com/dotnet/sdk/issues/1212

**Comment by glennawatson** [comment, marked as answer]
There is no global flag to turn the Refit source generator off, and you do not need one. The generator only emits an implementation for interfaces whose methods carry Refit HTTP attributes (Get, Post, etc.). It does not generate anything for arbitrary interfaces.

If you are seeing broken/partial generated code, it is because some of those interfaces have Refit attributes on their methods, so Refit tries to implement them. To stop that:

- Remove the Refit HTTP attributes from the interfaces you do not want Refit to implement, or
- Move the interfaces that are produced by your other source generator into a separate project/assembly that does not reference Refit, or
- Keep the Refit-attributed interfaces separate from the partial-extension ones.

Refit cannot selectively co-generate with another generator that targets the same interface, because both would emit members for the same type. Separation by attribute or by project is the supported way to control what Refit generates.

If you have a concrete repro where an interface with no Refit attributes still gets generated, please open an issue with a minimal sample.
